fedimint-api/src/tiered_multi.rs

```diff
@@ -78,35 +86,49 @@ impl<T> TieredMulti<T> {
         tier_eq && per_tier_eq
     }
 
-    pub fn iter_tiers(&self) -> impl Iterator<Item = (&Amount, &Vec<T>)> {
+    /// Returns an borrowing iterator
+    pub fn iter(&self) -> impl Iterator<Item = (&Amount, &Vec<T>)> {
         self.0.iter()
     }
 
-    // Note: order of the elements is important here: from lowest tiers to highest, then in order of elements in the Vec
+    /// Returns an iterator over every `(Amount, &T)`
+    ///
+    /// Note: The order of the elements is important:
+    /// from the lowest tier to the highest, then in order of elements in the Vec
     pub fn iter_items(&self) -> impl Iterator<Item = (Amount, &T)> + DoubleEndedIterator {
+        // Note: If you change the method implementation, make sure that the returned order of the
+        // elements stays consistent.
         self.0
             .iter()
             .flat_map(|(amt, coins)| coins.iter().map(move |c| (*amt, c)))
     }
-    // Note: order of the elements is important here: from lowest tiers to highest, then in order of elements in the Vec
+
+    /// Returns an consuming iterator over every `(Amount, T)`
+    ///
+    /// Note: The order of the elements is important:
+    /// from the lowest tier to the highest, then in order of elements in the Vec
     pub fn into_iter_items(self) -> impl Iterator<Item = (Amount, T)> + DoubleEndedIterator {
+        // Note: If you change the method implementation, make sure that the returned order of the
+        // elements stays consistent.
         self.0
             .into_iter()
             .flat_map(|(amt, coins)| coins.into_iter().map(move |c| (amt, c)))
     }
 
-    /// Returns the max number of notes for any given denomination tier
-    pub fn max_tier_len(&self) -> usize {
+    /// Returns the maximum length of all tier vectors
+    pub fn max_value_len(&self) -> usize {
```

Comment:
Now that I look at it: comment doesn't fit the name well.

`longest_tier_len` maybe?

Reply:
yeah, not the best name. how about `max_len()` or `max_vec_len()` 

I'll rewrite the comment, too. something like: `Returns the length of the longest vector of all tiers`

Reply:
Seems to me like `longest` will have to be there in the name. `max` is kind of vague - is it a lenght of "max" tier (highest amount)?

Reply:
mh, ok. on second thought, I like `longest_tier_len` -> give me the len of the longest tier :+1: